### ISSUE TITLE
[Bug ] ~4,200 unexpected crash records were pushed to VZ by CRIS on 2021-01-11

### ISSUE THREAD

**Issue by frankhereford** [cityofaustin/atd-data-tech#8140]
### TLDR

CRIS gave us another set of crash records and caused us to download a bunch of CR3s we shouldn't have had to. It's unknown what caused this. I propose we remove old, crufty queries from our query account.

### Situation

This morning, in the course of downloading CR3s, I observed that we had an extra set of CRIS export CSVs delivered to the VZ SFTP endpoint in addition to the normal daily set of CSVs we expect. The additional data set contained crashes from around November 1st, 2021 through today. This is confirmed via examination of the archive of processed CSV files.

Our system processed these crashes, which caused that set of crashes to have the record of having downloaded CR3s for those crashes to be reset. As a result, I am running the CR3 downloading program repeatedly to get those CR3s (re-)downloaded. Until this completes, the `Download CR3` button on the crashes will be missing for those crashes. At the time of writing, about 2K crashes remain, and are being processed about 1 per second. We should be restored to nominal operation in an hour or so.

I have not been able to figure out what caused this to happen. Yesterday, I did login to CRIS using the `transportation.data@austintexas.gov` account, which is used to manage our stored data requests. I did this to show @patrickm02L, via a screenshot, what I had meant when I said that I had prepended "FLH" to the query I made at the beginning of the year to carry us through 2022. However, I am exceptionally sure that I did nothing except login, take a screenshot of the welcome screen which lists queries and then close the window -- nothing that should have caused a query to be run. 

However, it does appear that either that action, another unrelated action by either another team member, or something internal to CRIS did cause one of our old, completed backfill queries to run and deliver us data. Interestingly, we didn't get /all/ of our old queries run, just one of them.

### Proposed Action

I believe we would benefit from clearing out old queries which we no longer expect to run, leaving just the recurring query which we get data from daily. @patrickm02L & @mateoclarke - what do you all think about doing that? 

I don't have any concrete information that suggests that this would have prevented this from happening, it just strikes me as something we can do to potentially lessen the chance of this happening again in the future.

I look forward to your thoughts - thank you.

**Comment by frankhereford** [comment]
This may be the cause. As per the "Important Messages from TxDOT" messages seen upon logging into CIRS:

> 1/11/2022 9:23 AM
> CRIS Release 23.0.5., deployed Sunday (1/9/22), included spatial layer and roadway data updates to CRIS. This requires all records to be reprocessed, with records being pulled in and out of the system 24/7 over the next 4 weeks. Data analysis during this time may produce inaccurate results.

**Comment by patrickm02L** [comment]
> This may be the cause. As per the "Important Messages from TxDOT" messages seen upon logging into CRIS:
> 
> > 1/11/2022 9:23 AM
> > CRIS Release 23.0.5., deployed Sunday (1/9/22), included spatial layer and roadway data updates to CRIS. This requires all records to be reprocessed, with records being pulled in and out of the system 24/7 over the next 4 weeks. Data analysis during this time may produce inaccurate results.

I have a feeling you're right.

**Comment by frankhereford** [comment]
All CR3 files have been downloaded for the affected set of crashes. Nominal operation appears to be restored for the VZE.

While, the impacts of this particular event have been cleared, I am going to leave this issue open until a team consensus is reached on what to do with old queries in the CRIS account.

**Comment by frankhereford** [comment]
Thanks for your thoughts Mateo. No worries on the email update - I would have done the same thing. 

I will close the issue, and I look forward to catching up with you on any next steps, and I'll bring it up today in our planning meeting to ensure that we have everyone's agreement on removing those old queries. Assuming we're all in agreement, I'll be happy to take care of that at the end of the day.

Thanks again!

**Comment by frankhereford** [comment]
Following up - we did get another extra set of CSVs last night, which started this cycle again of needing to (re-)download CR3s for ~8K crashes today, which is in progress currently. 

Based on the above conversation and a check-in with @mateoclarke, I have logged into the CRIS Extract Account (username: transportation.data@austintexas.gov) and removed all but the pair of data requests which are operating to provide our ongoing daily crash data from CRIS.

I have logged into the server which operates the SFTP endpoint that CRIS uploads the CSVs to, and there are no unprocessed CSVs on there at this moment. There is, however, a single set of CSV files in the S3 bucket, `atd-vision-zero-database/csv-extracts-sftp/` waiting to be processed the next time Airflow runs the DAG, which is scheduled for tonight @ 2AM. By my read, this is the system operating as expected and normally again.
